### PR TITLE
Enable the Equals, NoNeedForMonad, and Overloading warts.

### PR DESCRIPTION
**Pull request by sellout**
Some are still disabled at the project level.

### CODE REVIEW

**Comment by sellout** [comment]
Looks like puffnfresh/wartremover#268 is happening when we have coverage on.

**Comment by drostron** [comment]
I'm generally enthusiastic and in favor of additional wart enablement, but am a bit cautious at the moment due to the traversal per wart cost and the slow compile times we've been dealing with lately. It would be great to have some performance insight. Ideally the cost is low?

File: connector/src/main/scala/quasar/qscript/MapFunc.scala [hunk 1]
```diff
@@ -106,6 +106,9 @@ object MapFunc {
   object ConcatArraysN {
     private implicit def implicitPrio[F[_], G[_]]: Inject[G, Coproduct[F, G, ?]] = Inject.rightInjectInstance
 
+
```
Comment (sellout) [inline connector/src/main/scala/quasar/qscript/MapFunc.scala]: Well, that’s interesting …

**Comment by drostron** [comment]
Great to see a few `==` and `!=` going away.

File: foundation/src/main/scala/quasar/fp/package.scala [hunk 2]
```diff
@@ -317,36 +317,24 @@ trait SKI {
...
-  def κ[A, B](x: B): A => B                                 = _ => x
-  def κ[A, B, C](x: C): (A, B) => C                         = (_, _) => x
-  def κ[A, B, C, D](x: D): (A, B, C) => D                   = (_, _, _) => x
-  def κ[A, B, C, D, E, F, G](x: G): (A, B, C, D, E, F) => G = (_, _, _, _, _, _) => x
+  /** A shorter name for the constant function of 1, 2, 3, or 6 args.
+    * NB: the argument is eager here, so use `_ => ...` instead if you need it
+    *     to be thunked.
+    */
+  def κ[A, B](x: B): A => B                                  = _ => x
+  def κ2[A, B, C](x: C): (A, B) => C                         = (_, _) => x
```
Comment (drostron) [inline foundation/src/main/scala/quasar/fp/package.scala]: Of course, that wart escaped the tab title width 😉 .

**Comment by drostron** [comment]
LGTM. Will merge upon a ✅ build.

**Comment by drostron** [comment]
Oh, needs that comment update first too.

**Comment by drostron** [comment]
Did you have ideas for the build failure with coverage on other than disabling `NoNeedForMonad`?

**Comment by sellout** [comment]
Just wait for a patch for the Wart Remover bug – but we can disable it only on the projects that trigger that failure, at least.

**Comment by sellout** [comment]
Updated – hopefully a green build this time around.

**Comment by drostron** [comment]
A merge from master will likely address the build failures.

**Pull request merged**